### ISSUE TITLE
Errors/Warnings when uninstalling real-vnc

### ISSUE THREAD

**Issue by onlynone** [caskroom/homebrew-cask#9761]
When running `brew cask uninstall real-vnc` I get:

    ==> Running uninstall process for real-vnc; your password may be necessary
    ==> Running uninstall script /Applications/RealVNC/Advanced.localized/Uninstall VNC Server.app/Contents/Resources/uninstaller.sh
    Password:
    ==> Forgot package 'com.realvnc.vncserver.pkg' on '/'.
    ==> No matching processes were found
    ==> ls: /Applications/RealVNC/Advanced: No such file or directory
    ==> /Applications/RealVNC/Advanced.localized/Uninstall VNC Server.app/Contents/Resources/uninstallCommon.sh: line 29: ./removeFwExc.sh: No such file or directory
    ==> No receipt for 'com.realvnc.VNCServer.pkg' found at '/'.
    ==> Forgot package 'com.realvnc.vncviewer.pkg' on '/'.
    ==> ls: /Applications/RealVNC/Advanced: No such file or directory
    ==> No receipt for 'com.realvnc.VNCViewer.pkg' found at '/'.

I don't know if that's an error or not...

**Comment by radeksimko** [comment]
Hi, this is a known issue which I actually mentioned while adding the cask.
https://github.com/caskroom/homebrew-cask/pull/9579

TL;DR It's a bug in Real VNC, all reported back to them including suggested solutions.

The last response to my [ticket](https://gist.github.com/radeksimko/8aef85a6acde5475a888) I have from the support (8 days ago) is following:

> Hi Radek
> 
> I forwarded your comments to our Engineering team and they have asked me to thank you for your suggestions. They agree with both points.

> 1. A bug has been logged in our tracking system to fix the path issue in a future release.

> 2. Will be fixed in the script by:

> rmdir /Applications/RealVNC/Advanced* 2>/dev/null

> Thanks again for taking the time to make these suggestions.

The caveat for user (which exists even for users installing it outside brewcask manually) is that RealVNC apparently add a rule at some point to OSX firewall and doesn't remove it during the uninstallation process, other than that, the uninstallation works well.

We may add a `caveats` section mentioning this until they fix it?

**Comment by onlynone** [comment]
I think having `caveats` would be a good idea. Are they reported upon install or uninstall (or both)? If it's only reported that these are known issues during install, I probably still would have been confused when performing the uninstall.

Is there a way to programmatically remove the firewall rule on uninstall until they fix it?

**Comment by radeksimko** [comment]
> Are they reported upon install or uninstall (or both)?

I believe these are only reported during installation, but it might be nice to introduce something like `uninstall_caveats` or something similar? What do you think @caskroom/maintainers ? :)

> Is there a way to programmatically remove the firewall rule on uninstall until they fix it?

I guess there should be, I just wasn't looking into it too much and probably won't in the nearest future as I wanted to invest more time into solving some issues with Java/IntelliJ casks we have.

That said, if you'll come up with a pull request, I'll be happy to test it & merge it. :wink:

**Comment by rolandwalker** [comment]
+1 to `uninstall_caveats`.

Though there should be a way to manipulate the firewall rules from the CLI.  Apple published a great [comprehensive guide](http://manuals.info.apple.com/MANUALS/0/MA409/en_US/Command_Line_Admin_v10.5.pdf) on that sort of thing, though it is now out of date.  A good place to start sometimes.

**Comment by radeksimko** [comment]
@rolandwalker It's also worth mentioning, that the implementation for `uninstall :script` could be revised a little bit to support **array of commands**, so I don't have to use combination of `:early_script` and `:script` to do two things that are on the very same level. It's not a scalable approach as I can't imagine a clean way for adding third uninstall script.

After all I might be missing an obvious approach for this problem... :question:

**Comment by rolandwalker** [comment]
Don't we support repeating `uninstall :script` multiple times?  We should.

**Comment by vitorgalvao** [comment]
Seems to no longer be an issue. Worked fine in the latest version.
